### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by EidHachem**
## In this PR I implemented the following:

- Created a method called current_user in ApplicationController to make current user data available to all controllers.
        - It returns the first user from the database.
- Created forms to perform the following functions:
        - Creates a Post on behalf of the current_user.
        - Create a comment on behalf of the current_user.
-Allowed Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,9 +1,46 @@
...
+
+  def addlike
+    @user = current_user
+    @post = Post.find(params[:id])
+    @like = @post.likes.new
+    @like.post_id = @post.id
+    @like.author_id = @user.id
+    @like.save
+    redirect_to "/user/#{current_user.id}/post/#{@post.id}"
+  end
```
Comment (Deepakdanger) [inline app/controllers/posts_controller.rb]: - [x] There should be a `comments` and `likes` controller with the `create` method where you can create the comments and likes just like the `posts` controller here. The rails are designed in such a way that they will talk easily with the controller, views, and models.

File: config/routes.rb [hunk 2]
```diff
@@ -4,6 +4,10 @@
   # Defines the root path route ("/")
   root "users#index"
   get '/user/:id', to: 'users#show'
+  get '/post/new', to: 'posts#new'
+  post '/post/new', to: 'posts#create'
+  post '/addcomment/:id', to: "posts#addcomment"
   get '/user/:id/post/', to: 'posts#posts'
   get '/user/:id/post/:id', to: 'posts#post'
+  get 'addlike/:id', to: "posts#addlike"
```
Comment (Deepakdanger) [inline config/routes.rb]: - [x] After including the comments and likes controller, Also update the routes.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @EidHachem, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @EidHachem  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Made changes as requested by previous reviewer 👍 
- Linter checks passing ✔️ 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**